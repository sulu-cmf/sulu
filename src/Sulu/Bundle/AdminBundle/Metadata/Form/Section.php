<?php

/*
 * This file is part of Sulu.
 *
 * (c) MASSIVE ART WebServices GmbH
 *
 * This source file is subject to the MIT license that is bundled
 * with this source code in the file LICENSE.
 */

namespace Sulu\Bundle\AdminBundle\Metadata\Form;

class Section extends Item
{
    /**
     * @var Item[]
     */
    protected $items;

    protected $type = 'section';

    /**
     * @return Item[]
     */
    public function getItems(): array
    {
        return $this->items;
    }

    public function addItem(Item $item): void
    {
        $this->items[$item->getName()] = $item;
    }
}
