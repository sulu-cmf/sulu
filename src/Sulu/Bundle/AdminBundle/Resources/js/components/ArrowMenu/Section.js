// @flow
import React from 'react';
import type {ChildrenArray, Element} from 'react';
import sectionStyles from './section.scss';

type Props = {
    children?: ChildrenArray<Element<*>>,
    title?: string,
};

export default class Section extends React.PureComponent<Props> {
    render() {
        const {
            children,
            title,
        } = this.props;

        return (
            <div className={sectionStyles.section}>
                {title &&
                    <div className={sectionStyles.title}>{title}</div>
                }
                <div className={sectionStyles.children}>
                    {children}
                </div>
            </div>
        );
    }
}
