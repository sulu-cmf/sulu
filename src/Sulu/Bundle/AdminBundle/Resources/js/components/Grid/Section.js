// @flow
import React from 'react';
import type {Element} from 'react';
import classNames from 'classnames';
import type {BaseItemProps} from './types';
import BaseItem from './BaseItem';
import Item from './Item';
import sectionStyles from './section.scss';

type Props = BaseItemProps & {
    children: Element<typeof Item | typeof Section>,
    className?: string,
};

export default class Section extends React.PureComponent<Props> {
    static defaultProps = {
        size: 12,
        spaceBefore: 0,
        spaceAfter: 0,
    };

    render() {
        const {
            children,
            className,
            ...others
        } = this.props;

        const sectionClass = classNames([
            sectionStyles.section,
            className,
        ]);

        return (
            <BaseItem {...others} className={sectionClass}>
                {children}
            </BaseItem>
        );
    }
}
