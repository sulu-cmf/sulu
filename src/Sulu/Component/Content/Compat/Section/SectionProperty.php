<?php
/*
 * This file is part of the Sulu CMS.
 *
 * (c) MASSIVE ART WebServices GmbH
 *
 * This source file is subject to the MIT license that is bundled
 * with this source code in the file LICENSE.
 */

namespace Sulu\Component\Content\Section;

use JMS\Serializer\Context;
use JMS\Serializer\JsonDeserializationVisitor;
use JMS\Serializer\JsonSerializationVisitor;
use Sulu\Component\Content\Property;
use Sulu\Component\Content\PropertyInterface;
use JMS\Serializer\Annotation\Type;
use JMS\Serializer\Annotation\HandlerCallback;

/**
 * defines a section for properties
 * @package Sulu\Component\Content
 */
class SectionProperty extends Property implements SectionPropertyInterface
{
    /**
     * properties managed by this block
     * @var PropertyInterface[]
     * @Type("array<Sulu\Component\Content\Property>")
     */
    private $childProperties = array();

    /**
     * @param string $name
     * @param array $metadata
     * @param string $col
     */
    public function __construct($name, $metadata, $col)
    {
        parent::__construct($name, $metadata, 'section', false, false, 1, 1, array(), array(), $col);
    }

    /**
     * {@inheritdoc}
     */
    public function getChildProperties()
    {
        return $this->childProperties;
    }

    /**
     * {@inheritdoc}
     */
    public function addChild(PropertyInterface $property)
    {
        $this->childProperties[] = $property;
    }

    /**
     * @HandlerCallback("json", direction = "serialization")
     */
    public function serializeToJson(JsonSerializationVisitor $visitor, $data, Context $context)
    {
        return parent::serializeToJson($visitor, $data, $context);
    }

    /**
     * @HandlerCallback("json", direction = "deserialization")
     */
    public function deserializeToJson(JsonDeserializationVisitor $visitor, $data, Context $context)
    {
        return parent::deserializeToJson($visitor, $data, $context);
    }
}
