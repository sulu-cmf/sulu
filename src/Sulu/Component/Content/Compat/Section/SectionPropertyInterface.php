<?php
/*
 * This file is part of the Sulu CMS.
 *
 * (c) MASSIVE ART WebServices GmbH
 *
 * This source file is subject to the MIT license that is bundled
 * with this source code in the file LICENSE.
 */

namespace Sulu\Component\Content\Compat\Section;

use Sulu\Component\Content\Compat\PropertyInterface;

/**
 * Defines a section for properties
 */
interface SectionPropertyInterface extends PropertyInterface
{
    /**
     * returns a list of properties managed by this block.
     *
     * @return PropertyInterface[]
     */
    public function getChildProperties();

    /**
     * add a child to section.
     *
     * @param PropertyInterface $property
     */
    public function addChild(PropertyInterface $property);
}
