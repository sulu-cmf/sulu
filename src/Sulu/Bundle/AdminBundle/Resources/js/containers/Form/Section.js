// @flow
import React from 'react';
import type {ChildrenArray, Element} from 'react';
import {computed} from 'mobx';
import {observer} from 'mobx-react';
import jexl from 'jexl';
import Form from '../../components/Form';
import conditionDataProviderRegistry from './registries/conditionDataProviderRegistry';
import FormInspector from './FormInspector';
import Field from './Field';
import type {SchemaEntry} from './types';

type Props = {|
    children: false | ChildrenArray<?Element<typeof Field | typeof Section>>,
    data: Object,
    formInspector: FormInspector,
    name: string,
    schema: SchemaEntry,
|};

@observer
class Section extends React.Component<Props> {
    @computed get conditionData() {
        const {data, formInspector} = this.props;
        const {locale, metadataOptions, options} = formInspector;

        return conditionDataProviderRegistry.getAll().reduce(
            function(data, conditionDataProvider) {
                return {...data, ...conditionDataProvider(data, options, metadataOptions)};
            },
            {...data, __locale: locale}
        );
    }

    @computed get visible() {
        const {schema} = this.props;

        if (!schema.visibleCondition) {
            return true;
        }

        return jexl.evalSync(schema.visibleCondition, this.conditionData);
    }

    render() {
        if (!this.visible) {
            return null;
        }

        const {children, name, schema} = this.props;
        const {colSpan, label} = schema;

        return (
            <Form.Section colSpan={colSpan} key={name} label={label}>
                {children}
            </Form.Section>
        );
    }
}

export default Section;
