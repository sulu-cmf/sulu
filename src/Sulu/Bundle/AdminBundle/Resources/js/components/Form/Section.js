// @flow
import React from 'react';
import type {Node} from 'react';
import Divider from '../Divider';
import Grid from '../Grid';
import type {Size} from '../Grid';
import gridStyles from './grid.scss';

type Props = {|
    children: Node,
    label?: string,
    size: Size,
|};

export default class Section extends React.Component<Props> {
    static defaultProps = {
        size: 12,
    };

    render() {
        const {children, label, size} = this.props;

        return (
            <Grid.Section className={gridStyles.gridSection} size={size}>
                {label &&
                    <Grid.Item size={12}>
                        <Divider>
                            {label}
                        </Divider>
                    </Grid.Item>
                }
                {children}
            </Grid.Section>
        );
    }
}
