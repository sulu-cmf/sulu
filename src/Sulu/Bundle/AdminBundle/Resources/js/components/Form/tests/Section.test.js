// @flow
import React from 'react';
import {render} from 'enzyme';
import Section from '../Section';

test('Render section with given size', () => {
    expect(render(
        <Section label="Test">
            <p>Test</p>
        </Section>
    )).toMatchSnapshot();
});

test('Render section without label', () => {
    expect(render(
        <Section>
            <div>Test</div>
        </Section>
    )).toMatchSnapshot();
});
