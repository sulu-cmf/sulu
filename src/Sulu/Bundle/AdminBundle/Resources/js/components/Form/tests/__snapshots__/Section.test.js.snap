// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render section with given size 1`] = `
<div
  class="section gridSection size size-12 space-before-0 space-after-0"
>
  <div
    class="item size size-12 space-before-0 space-after-0"
  >
    <div
      class="divider"
    >
      Test
    </div>
  </div>
  <p>
    Test
  </p>
</div>
`;

exports[`Render section without label 1`] = `
<div
  class="section gridSection size size-12 space-before-0 space-after-0"
>
  <div>
    Test
  </div>
</div>
`;
